File: Rotated Array Min.java
// Time Complexity : log(N)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach

class Solution {
    public int findMin(int[] nums) {
        int start=0;
        int end=nums.length-1;
        
         if(nums.length==1 || nums[start]<nums[end]) return nums[start];
        while(start<=end){
            int mid=start+(end-start)/2;
           if(start==mid && nums[mid+1]<nums[mid]) return nums[mid+1];
            if(nums[start]>nums[mid]){
                end=mid;
            }else{
                start=mid;
            }
        }
        
        return nums[0];
    }
}

File: rotatedArrayMin.py
class Solution:
    def findMin(self, nums):

        low = 0
        high = len(nums) - 1

        while low <= high:
            if(nums[low] < nums[high]):
                return nums[low]

            med = low + (high - low) // 2

            if ((med == low or nums[med-1] > nums[med]) and (med == high or nums[med+1] > nums[med])):
                return nums[med]

            elif nums[low] <= nums[med]:
                low = med + 1
            else:
                high = med - 1

      

nums = [3, 4, 5, 1, 2]
res = Solution()
output = res.findMin(nums)
print(output)
